src/server/enterprise/server/api_server.go

```diff
@@ -330,3 +345,95 @@ func (a *apiServer) Deactivate(ctx context.Context, req *ec.DeactivateRequest) (
 
 	return &ec.DeactivateResponse{}, nil
 }
+
+func (a *apiServer) Pause(ctx context.Context, req *ec.PauseRequest) (resp *ec.PauseResponse, retErr error) {
+	if req.Namespace == "" {
+		req.Namespace = "default"
+	}
+	if err := a.setPauseState(ctx, req.Namespace, true); err != nil {
+		return nil, err
+	}
+	if err := scaleDownWorkers(ctx, a.env.GetKubeClient(), req.Namespace); err != nil {
+		return nil, errors.EnsureStack(err)
+	}
+	if err := rollPachd(ctx, a.env.GetKubeClient(), req.Namespace); err != nil {
+		return nil, errors.EnsureStack(err)
+	}
+
+	return &ec.PauseResponse{}, nil
```

Comment:
maybe a naive question - but is it possible for the pod to restart before we return the response?